tests/test_product.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+from buy_together_app.models import Product
+from buy_together_app.tests import TEST_DATA
+
+QR = "Q5o76MbdiNbXprNEnHfpcGWFp1CMF8XY"
+NAME = "Banana"
+DESCRIPTION = "Good!"
+
+
+@pytest.fixture
+def product0():
+    return Product(qr_code=QR, product_name=NAME, description=DESCRIPTION)
+
+
+@pytest.fixture
+def saved_product0(product0):
+    assert Product.create_product(product0)
+    return product0
+
+
+class TestProductModel:
+    @pytest.mark.django_db()
+    def test_save_product(self, saved_product0):
+        Product.delete_product(saved_product0)
+        Product.create_product(saved_product0)
+        assert saved_product0 in Product.get_all_products()
+
+    @pytest.mark.django_db()
+    def test_delete_product(self, saved_product0):
+        assert saved_product0 in Product.get_all_products()
+        Product.delete_product(saved_product0)
+        assert saved_product0 not in Product.get_all_products()
+
+    @pytest.mark.django_db()
+    def test_filter_by_qr(self, saved_product0):
+        assert list(Product.get_all_products()) == list(Product.filter_qr(""))
+        assert [saved_product0] == list(Product.filter_qr(saved_product0.qr_code))
+
+    @pytest.mark.django_db()
+    def test_filter_by_product_name(self, saved_product0):
+        assert list(Product.get_all_products()) == list(Product.filter_name(""))
+        assert saved_product0 in list(Product.filter_name(saved_product0.product_name))
+
+    @pytest.mark.django_db()
+    def test_filter_by_description(self, saved_product0):
+        assert list(Product.get_all_products()) == list(Product.filter_description(""))
+        assert saved_product0 in Product.filter_description(saved_product0.description)
+
+    @pytest.mark.django_db()
+    def test_get_all_products(self):
+        """
+            Suppose we have empty database and we need to add to it 3 apples
+            we need to check if the all products are actually apples and same
+            with same qr_codes
+        """
+        prods = []
+        for qr, name, desc in TEST_DATA:
+            temp = Product(qr, name, desc)
+            Product.create_product(temp)
+            prods.append(temp)
```

Comment:
Can make it more pythonic:
```suggestion python
prods = [Product(*td) for td in TEST_DATA]
for p in prods:
    Product.create_product(p)
```
or 
```suggestion python
prods = [Product(*td) for td in TEST_DATA]
list(map(Product.create_product, prods))
```

Reply:
Done

Reply:
I still see the same code